spec/smartcar/e2e/vehicle_spec.rb

```diff
@@ -5,181 +5,191 @@
-      result = @vehicle.fuel
-      expect(result.instance_of?(Smartcar::Fuel)).to eq(true)
-      expect(result.percent_remaining).not_to be_nil
-      expect(result.amount_remaining).not_to be_nil
-      expect(result.meta).not_to be_nil
-      expect(result.range).not_to be_nil
+    describe '#tire_pressure' do
+      it 'should return an tire_pressure object' do
+        result = @vehicle.tire_pressure
+        %I[back_left front_left back_right front_right].each do |attribute|
+          expect(result[attribute].instance_of?(Float)).to eq(true)
+        end
+        expect(result.meta.request_id.length).to eq(36)
+        expect(result.meta.unit_system).to eq('metric')
+        expect(result.meta.data_age.is_a?(DateTime)).to eq(true)
+      end
     end
-  end
 
-  describe '#location' do
-    it 'should return an location object' do
-      result = @vehicle.location
-      expect(result.instance_of?(Smartcar::Location)).to eq(true)
-      expect(result.latitude).not_to be_nil
-      expect(result.meta).not_to be_nil
-      expect(result.longitude).not_to be_nil
+    describe '#vin' do
+      it 'should return an vin string' do
+        result = @vehicle.vin
+        expect(result.vin.length).to eq(17)
+        expect(result.meta.request_id.length).to eq(36)
+      end
     end
-  end
 
-  describe '#permissions' do
-    it 'should return an permissions object' do
-      result = @vehicle.permissions
-      expect(result.instance_of?(Smartcar::Permissions)).to eq(true)
-      expect(result.meta).not_to be_nil
-      expect(result.permissions).not_to be_nil
+    describe '#odometer' do
+      it 'should return an odometer object' do
+        result = @vehicle.odometer
+        expect(result.distance.instance_of?(Float)).to eq(true)
+        expect(result.meta.request_id.length).to eq(36)
+        expect(result.meta.unit_system).to eq('metric')
+        expect(result.meta.data_age.is_a?(DateTime)).to eq(true)
+      end
     end
-  end
 
-  describe '#tire_pressure' do
-    it 'should return an tire_pressure object' do
-      result = @vehicle.tire_pressure
-      expect(result.instance_of?(Smartcar::TirePressure)).to eq(true)
-      expect(result.meta).not_to be_nil
-      expect(result.back_left).not_to be_nil
-      expect(result.front_left).not_to be_nil
-      expect(result.back_right).not_to be_nil
-      expect(result.front_right).not_to be_nil
+    describe '#batch - success' do
+      context 'with valid attributes' do
+        it 'should return hash of objects with attribute requested as keys' do
+          attributes = ['/charge', '/battery', '/odometer']
+          result = @vehicle.batch(attributes)
+          expect(result.instance_of?(OpenStruct)).to eq(true)
+          expect(result.charge.instance_of?(OpenStruct)).to eq(true)
+          expect(result.charge.is_plugged_in?).not_to be_nil
+          expect(result.charge.state).not_to be_nil
+          expect(result.charge.meta).not_to be_nil
```

Comment:
```suggestion
          expect(result.charge.meta).not_to be_nil
          expect(result.charge.meta.request_id.length).to eq(36)
```

just to make sure meta is built correctly for batch (testing the case of headers being merged)